app/models/default_group_cover.rb

```diff
@@ -0,0 +1,14 @@
+class DefaultGroupCover < ActiveRecord::Base
+  has_attached_file :cover_photo
+  validates_attachment :cover_photo, content_type: { content_type: ["image/jpg", "image/jpeg", "image/png", "image/gif"] }
+
+  def self.sample
+    all.order('random()').first
+  end
+
+  def self.store(file)
+    file = open(file) if file.is_a? String
```

Comment:
is this a IO open? If so you might need to close it. Also if so, use a do |file| end so you don't need to close it manually.